### PR TITLE
[4.10-4.14] Add missing GCP Workload Identity jobs to Install and Upgrade CI

### PR DESCRIPTION
**Pull request by jianli-wei**
https://issues.redhat.com/browse/OCPQE-15522

**Updated for 4.14 and 4.14-upgrade-from-stable-4.13:** 
Old Profile | New Profile | Notes
-- | -- | -- 
gcp-ipi-disconnected-private-p2-f14 | gcp-ipi-disconnected-private-workload-identity-p2-f14 | 4.14-upgrade-from-stable-4.13
gcp-ipi-realtime-kernel-fips-p2-f28, gcp-ipi-workload-identity-fips-p1-f14 | gcp-ipi-workload-identity-rt-fips-p1-f14 | 4.14-upgrade-from-stable-4.13
gcp-ipi-disconnected-private-p3-f28 || 4.14
gcp-ipi-realtime-kernel-fips-p2-f14 | gcp-ipi-workload-identity-rt-fips-p2-f14 | 4.14
gcp-ipi-realtime-kernel-fips-p2-f28-destructive, gcp-ipi-workload-identity-fips-p1-f28-destructive | gcp-ipi-workload-identity-rt-fips-p3-f28-destructive | 4.14

_P.S. Renaming "realtime-kernel" as "rt", so that `gcp-ipi-workload-identity-rt-fips-p1-f28-destructive` stays within 61 characters (which seems the upper limit)._
```
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.14__amd64-nightly-4.14-upgrade-from-stable-4.13.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
- as: gcp-ipi-workload-identity-rt-fips-p1-f14
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml
- as: gcp-ipi-disconnected-private-p3-f28-destructive
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
- as: gcp-ipi-disconnected-workers-rhel8-p2-f14
- as: gcp-ipi-disconnected-workers-rhel8-p2-f28-destructive
- as: gcp-ipi-workload-identity-rt-fips-p2-f14
- as: gcp-ipi-workload-identity-rt-fips-p3-f28-destructive
[jiwei@jiwei openshift-tests-private]$ 
```

**Updated for 4.13, 4.13-upgrade-from-stable-4.12, 4.13-upgrade-from-stable-4.13 and stable-4.13-upgrade-from-stable-4.12:** 
Old Profile | New Profile | Notes
-- | -- | -- 
gcp-ipi-disconnected-private-p2-f14 | gcp-ipi-disconnected-private-workload-identity-p2-f14 | 4.13-upgrade-from-stable-4.12
gcp-ipi-realtime-kernel-fips-p2-f28, gcp-ipi-workload-identity-fips-p1-f14 | gcp-ipi-workload-identity-rt-fips-p1-f14 | 4.13-upgrade-from-stable-4.12
<none> | gcp-ipi-disconnected-private-workload-identity-p3-f28 | 4.13-upgrade-from-stable-4.13
gcp-ipi-disconnected-private-p3-f28 | | 4.13
gcp-ipi-realtime-kernel-fips-p2-f14 | gcp-ipi-workload-identity-rt-fips-p2-f14 | 4.13
gcp-ipi-realtime-kernel-fips-p2-f28-destructive, gcp-ipi-workload-identity-fips-p1-f28-destructive | gcp-ipi-workload-identity-rt-fips-p3-f28-destructive | 4.13
gcp-ipi-disconnected-private-p2-f14 | gcp-ipi-disconnected-private-workload-identity-p2-f14 | stable-4.13-upgrade-from-stable-4.12

```
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
- as: gcp-ipi-workload-identity-rt-fips-p1-f14
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.13.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p3-f28
- as: gcp-ipi-workload-identity-p3-f28
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml
- as: gcp-ipi-disconnected-private-p3-f28-destructive
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
- as: gcp-ipi-disconnected-workers-rhel8-p2-f14
- as: gcp-ipi-disconnected-workers-rhel8-p2-f28-destructive
- as: gcp-ipi-workload-identity-rt-fips-p2-f14
- as: gcp-ipi-workload-identity-rt-fips-p3-f28-destructive
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.13__amd64-stable-4.13-upgrade-from-stable-4.12.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
[jiwei@jiwei openshift-tests-private]$ 
```

**Updated for 4.12, 4.12__amd64-stable, 4.12-upgrade-from-stable-4.11, 4.12-upgrade-from-stable-4.12 and stable-4.12-upgrade-from-stable-4.11:**
Old Profile | New Profile | Notes
-- | -- | -- 
gcp-ipi-disconnected-private-p2-f14 | gcp-ipi-disconnected-private-workload-identity-p2-f14 | 4.12-upgrade-from-stable-4.11
gcp-ipi-workload-identity-p2-f14, gcp-ipi-realtime-kernel-fips-p2-f28 | gcp-ipi-workload-identity-rt-fips-p2-f14 | 4.12-upgrade-from-stable-4.11
<none> | gcp-ipi-disconnected-private-workload-identity-p2-f28 | 4.12-upgrade-from-stable-4.12
gcp-ipi-disconnected-private-p3-f28 | | 4.12
gcp-ipi-realtime-kernel-fips-p2-f14 | gcp-ipi-workload-identity-rt-fips-p2-f14 | 4.12
gcp-ipi-realtime-kernel-fips-p2-f28-destructive, gcp-ipi-workload-identity-fips-p3-f28-destructive | gcp-ipi-workload-identity-rt-fips-p3-f28-destructive | 4.12
gcp-ipi-disconnected-private-p2-f14 | gcp-ipi-disconnected-private-workload-identity-p2-f14 | stable-4.12-upgrade-from-stable-4.11
gcp-ipi-disconnected-private-p2-f14 | gcp-ipi-disconnected-private-workload-identity-p2-f14 | 4.12__amd64-stable

```
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-stable-4.11.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
- as: gcp-ipi-workload-identity-rt-fips-p2-f14
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-stable-4.12.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f28
- as: gcp-ipi-workload-identity-p2-f28
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml
- as: gcp-ipi-disconnected-private-p3-f28-destructive
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
- as: gcp-ipi-disconnected-workers-rhel8-p2-f14
- as: gcp-ipi-disconnected-workers-rhel8-p2-f28-destructive
- as: gcp-ipi-workload-identity-rt-fips-p2-f14
- as: gcp-ipi-workload-identity-rt-fips-p3-f28-destructive
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.12__amd64-stable-4.12-upgrade-from-stable-4.11.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.12__amd64-stable.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
[jiwei@jiwei openshift-tests-private]$ 
```

**Updated for 4.11, 4.11-upgrade-from-stable-4.10, 4.11-upgrade-from-stable-4.11 and 4.11__amd64-stable:**
Old Profile | New Profile | Notes
-- | -- | -- 
gcp-ipi-disconnected-private-p2-f28 | gcp-ipi-disconnected-private-workload-identity-p2-f28 | 4.11-upgrade-from-stable-4.10
gcp-ipi-realtime-kernel-fips-p2-f28, gcp-ipi-workload-identity-fips-p2-f28 | gcp-ipi-workload-identity-rt-fips-p2-f28 | 4.11-upgrade-from-stable-4.10
gcp-ipi-disconnected-private-p2-f360 | gcp-ipi-disconnected-private-workload-identity-p2-f360 | 4.11-upgrade-from-stable-4.11
gcp-ipi-realtime-kernel-fips-p2-f360, gcp-ipi-workload-identity-fips-p2-f360 | gcp-ipi-workload-identity-rt-fips-p2-f360 | 4.11-upgrade-from-stable-4.11
gcp-ipi-disconnected-private-p3-f28 | gcp-ipi-disconnected-private-workload-identity-p2-f28 | 4.11
gcp-ipi-realtime-kernel-fips-p2-f14, gcp-ipi-workload-identity-fips-p2-f14 | gcp-ipi-workload-identity-rt-fips-p2-f14 | 4.11
gcp-ipi-realtime-kernel-fips-p2-f28-destructive, gcp-ipi-workload-identity-fips-p3-f28-destructive | gcp-ipi-workload-identity-rt-fips-p3-f28-destructive | 4.11
gcp-ipi-disconnected-private-p2-f14 | gcp-ipi-disconnected-private-workload-identity-p2-f14 | 4.11__amd64-stable
```
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-stable-4.10.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f28
- as: gcp-ipi-workload-identity-rt-fips-p2-f28
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-stable-4.11.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f360
- as: gcp-ipi-workload-identity-rt-fips-p2-f360
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml
- as: gcp-ipi-disconnected-private-ovn-p2-f14
- as: gcp-ipi-disconnected-private-ovn-p2-f28-destructive
- as: gcp-ipi-disconnected-private-p3-f28-destructive
- as: gcp-ipi-disconnected-private-workload-identity-p2-f28
- as: gcp-ipi-disconnected-workers-rhel8-p2-f14
- as: gcp-ipi-disconnected-workers-rhel8-p2-f28-destructive
- as: gcp-ipi-workload-identity-rt-fips-p2-f14
- as: gcp-ipi-workload-identity-rt-fips-p3-f28-destructive
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.11__amd64-stable.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
[jiwei@jiwei openshift-tests-private]$ 
```

**Updated for 4.10, 4.10-upgrade-from-stable-4.9 and 4.10__amd64-stable:**
Old Profile | New Profile | Notes
-- | -- | -- 
gcp-ipi-disconnected-private-p2-f28 | gcp-ipi-disconnected-private-workload-identity-p2-f28 | 4.10-upgrade-from-stable-4.9
gcp-ipi-realtime-kernel-p2-f28 | gcp-ipi-workload-identity-rt-p2-f28 | 4.10-upgrade-from-stable-4.9
gcp-ipi-disconnected-private-p3-f28 | gcp-ipi-disconnected-private-workload-identity-p3-f28 | 4.10
gcp-ipi-realtime-kernel-fips-p2-f28, gcp-ipi-workload-identity-fips-p2-f28 | gcp-ipi-workload-identity-rt-fips-p2-f28 | 4.10
gcp-ipi-realtime-kernel-fips-p2-f28-destructive, gcp-ipi-workload-identity-fips-p3-f28-destructive | gcp-ipi-workload-identity-rt-fips-p3-f28-destructive | 4.10
gcp-ipi-disconnected-private-p2-f28 | gcp-ipi-disconnected-private-workload-identity-p2-f28 | 4.10__amd64-stable
```
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.10__amd64-nightly-4.10-upgrade-from-stable-4.9.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f28
- as: gcp-ipi-workload-identity-rt-p2-f28
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml
- as: gcp-ipi-disconnected-private-ovn-p2-f28
- as: gcp-ipi-disconnected-private-ovn-p2-f28-destructive
- as: gcp-ipi-disconnected-private-p3-f28-destructive
- as: gcp-ipi-disconnected-private-workload-identity-p3-f28
- as: gcp-ipi-disconnected-workers-rhel8-p2-f28
- as: gcp-ipi-disconnected-workers-rhel8-p2-f28-destructive
- as: gcp-ipi-workload-identity-rt-fips-p2-f28
- as: gcp-ipi-workload-identity-rt-fips-p3-f28-destructive
[jiwei@jiwei openshift-tests-private]$ grep -E "as: gcp-.*(workload|cco|sts|disconn|real|-rt)" openshift-openshift-tests-private-release-4.10__amd64-stable.yaml
- as: gcp-ipi-disconnected-private-workload-identity-p2-f28
[jiwei@jiwei openshift-tests-private]$ 
```

### CODE REVIEW

**Comment by jianli-wei** [comment]
@jianlinliu @gpei I updated for 4.14 (install & upgrade from 4.13), could you help review? Then I'm move on with other versions. Thanks!

**Comment by gpei** [comment]
@jianli-wei I'm ok with the changes

**Comment by jianlinliu** [comment]
@jianli-wei Personally I suggest to keep the same configuration in fresh install and upgrade ci, so that we can easily compare them among the jobs for easy analysis.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-stable-4.11.yaml [hunk 1]
```diff
@@ -433,20 +433,8 @@ tests:
...
-    env:
-      UPGRADE_CHECK_RUN_TAGS: '@upgrade-check and @4.11 and @gcp-ipi and not @fips'
-      UPGRADE_PRE_RUN_TAGS: '@upgrade-prepare and @4.11 and @gcp-ipi and not @fips'
-    test:
-    - ref: cucushift-installer-check
-    - chain: openshift-upgrade-qe-test
-    workflow: cucushift-installer-rehearse-gcp-ipi-cco-manual-workload-identity
-- as: gcp-ipi-disconnected-private-p2-f14
-  cron: 26 4 5,20 * *
+- as: gcp-ipi-disconnected-private-workload-identity-p2-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-stable-4.11.yaml:436]: In the source version - 4.11, there is no `gcp-ipi-disconnected-private-workload-identity-p2-f14` in openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml, are we going to add the same job in openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 2]
```diff
@@ -2806,30 +2793,19 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-gcp-ipi-proxy
-- as: gcp-ipi-realtime-kernel-fips-p2-f14
-  cron: 11 8 13,28 * *
+- as: gcp-ipi-rt-fips-p2-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2796]: this one is already turned into `gcp-ipi-workload-identity-rt-fips-p1-f14`, right?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 3]
```diff
@@ -2913,18 +2876,32 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-gcp-ipi-workers-rhel8-fips
-- as: gcp-ipi-workload-identity-fips-p1-f28-destructive
-  cron: 2 3 3 * *
+- as: gcp-ipi-workload-identity-rt-fips-p1-f28-destructive
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2879]: suggest to change `p1` to `p3`, and swap the order of `gcp-ipi-workload-identity-rt-fips-p1-f28-destructive` and `gcp-ipi-workload-identity-rt-fips-p2-f14`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-stable-4.11.yaml (hunk 1 above)
Comment (jianli-wei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-stable-4.11.yaml:436]: @jianlinliu sorry, just finished 4.14 - 4.12 (fresh install and upgrade), not yet touch 4.11 and 4.10, will do.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml (hunk 2 above)
Comment (jianli-wei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml]: It turned into `gcp-ipi-workload-identity-rt-fips-p2-f14` in 4.13.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly-4.10-upgrade-from-stable-4.9.yaml [hunk 4]
```diff
@@ -379,8 +379,8 @@ tests:
     - ref: cucushift-installer-check
     - chain: openshift-upgrade-qe-test
     workflow: cucushift-installer-rehearse-azure-upi
-- as: gcp-ipi-disconnected-private-p2-f28
-  cron: 13 19 3 * *
+- as: gcp-ipi-disconnected-private-workload-identity-p3-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly-4.10-upgrade-from-stable-4.9.yaml:382]: how about "p2"?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml [hunk 5]
```diff
@@ -1339,32 +1339,32 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-gcp-ipi-disconnected-private-ovn
-- as: gcp-ipi-disconnected-private-p3-f28
-  cron: 49 4 16 * *
+- as: gcp-ipi-disconnected-private-p3-f28-destructive
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml:1342]: suggest to update it on `p2`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-stable-4.10.yaml [hunk 6]
```diff
@@ -413,8 +413,8 @@ tests:
     - ref: cucushift-installer-check
     - chain: openshift-upgrade-qe-test
     workflow: cucushift-installer-rehearse-baremetalds-ipi-ovn-ipv4
-- as: gcp-ipi-disconnected-private-p2-f28
-  cron: 13 4 29 * *
+- as: gcp-ipi-disconnected-private-workload-identity-p3-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-stable-4.10.yaml:416]: suggest to update it on `p2`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-stable-4.10.yaml [hunk 7]
```diff
@@ -476,13 +463,14 @@ tests:
     - ref: cucushift-installer-check
     - chain: openshift-upgrade-qe-test
     workflow: cucushift-installer-rehearse-gcp-ipi-workers-rhel8
-- as: gcp-ipi-workload-identity-fips-p2-f28
-  cron: 19 14 7 * *
+- as: gcp-ipi-workload-identity-rt-fips-p3-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-stable-4.10.yaml:466]: suggest to update it on `p2`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml [hunk 8]
```diff
@@ -1513,32 +1513,32 @@ tests:
...
       TAG_VERSION: '@4.11'
-      TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
+      TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;Serial;Disruptive;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
     test:
-    - chain: openshift-e2e-test-qe
+    - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-gcp-ipi-disconnected-private
-- as: gcp-ipi-disconnected-private-p3-f28-destructive
-  cron: 7 7 16 * *
+- as: gcp-ipi-disconnected-private-workload-identity-p3-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml:1529]: suggest to update it on `p2`

**Comment by jianli-wei** [comment]
@jianlinliu @gpei I updated for 4.14 - 4.10, including fresh install and upgrade to these versions. When you have time, please help review, thanks!

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Comment by jianlinliu** [comment]
@jhou1 can you help review because @lxia would not be online today.

**Pull request merged**